### DISCUSSION TITLE
As a new user, I expect to be able to print the left side of assignments in breakpoints

### DISCUSSION THREAD

**Discussion by jcmorrow** [RecordReplay/devtools#4025] [Ideas]
I'll explain with a video capture:

https://user-images.githubusercontent.com/5903784/137429729-3fba0101-a503-4762-b785-d44c7e13cfb8.mp4

I have seen users flag this "bug" before, and I myself got confused by it at first, until realizing that our evaluation was actually *prior* to the assignment. Admittedly, it's not a bug in the sense that it was unintended, but it feels like either:

- We should communicate the breakpoint location more specifically (maybe column breakpoints will help?)
OR
- We should make the machine behave the way a user might expect on first glance, even though there are good reasons for it to behave otherwise

Or maybe there's a third option I haven't thought about. Whatever we choose, it feels like we risk giving a first impression that the tool is fundamentally broken when interacting with one of it most useful features.

**Comment by jonbell-lot23** [comment]
March 15, 2022:

This has always been a confusing part of Replay. Worth continuing to work on because I think it's a big hurdle for people learning the app.
